core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -65,12 +72,20 @@ public TermSuggestionBuilder(String name) {
      * tokens in the suggest text.
      * </ol>
      */
-    public TermSuggestionBuilder suggestMode(String suggestMode) {
+    public TermSuggestionBuilder suggestMode(@Nullable SuggestMode suggestMode) {
```

Comment:
The `DirectSpellcheckerSettings` contains some hard coded values which we can put into constants that both the builder and the settings use.  Is there a convention for where to put those constants - in builders or in these "settings" objects or elsewhere?